Nodes/ArduinoNode.cpp

```diff
@@ -0,0 +1,123 @@
+/****************************************************************************************
+ *
+ * File:
+ * 		ArduinoNode.cpp
+ *
+ * Purpose:
+ *		The Arduino node communicates with the arduino. Sends data about the pressure, rudder, sheet and battery.
+ *
+ *
+ * Developer Notes:
+ *
+ *
+ ***************************************************************************************/
+
+#include "ArduinoNode.h"
+#include "Messages/ArduinoDataMsg.h"
+#include "logger/Logger.h"
+
+// For std::this_thread
+#include <chrono>
+#include <thread>
+
+#define BLOCK_READ_SIZE 9
+#define BLOCK_I2C_ADDRESS_LOC 8
+
+#define DEFAULT_I2C_ADDRESS_PRESSURE 0x07
+
+#define ARDUINO_SENSOR_SLEEP_MS	100
+
+
+ArduinoNode::ArduinoNode(MessageBus& msgBus)
+: ActiveNode(NodeID::Arduino, msgBus), m_Initialised(false)
+{
+    
+}
+
+bool ArduinoNode::init()
+{
+	m_Initialised = false;
+
+    if(m_I2C.init(DEFAULT_I2C_ADDRESS_PRESSURE))
+	{
+		m_I2C.beginTransmission();
+
+        uint8_t block[BLOCK_READ_SIZE];
+        m_I2C.readBlock(block, BLOCK_READ_SIZE);
+		uint8_t deviceID = block[BLOCK_I2C_ADDRESS_LOC];
+
+		m_I2C.endTransmission();
+
+		// The Device reports the I2C address that is mentioned in the datasheet
+		if(deviceID == DEFAULT_I2C_ADDRESS_PRESSURE)
+		{
+			m_Initialised = true;
+		}
+		else
+		{
+			Logger::error("%s Failed to successfully read from the Arduino, check connection!", __PRETTY_FUNCTION__);
+		}
+	}
+	else
+	{
+		Logger::error("%s Failed to obtain I2C file descriptor", __PRETTY_FUNCTION__);
+	}
+
+	return m_Initialised;
+}
+
+void ArduinoNode::start()
+{
+	if(m_Initialised)
+	{
+		runThread(ArduinoThreadFunc);
+	}
+	else
+	{
+		Logger::error("%s Cannot start Aurdino thread as the node was not correctly initialised!", __PRETTY_FUNCTION__);
+	}
+}
+
+void ArduinoNode::processMessage(const Message* msg)
+{
+
+}
+
+void ArduinoNode::ArduinoThreadFunc(void* nodePtr)
+{
+    ArduinoNode* node = (ArduinoNode*)nodePtr;
+
+    Logger::info("Arduino thread started");
```

Comment:
If the node failed to initialize, this thread shouldn't run, put a check in here.